spec/defines/openssl_export_pem_cert_spec.rb

```diff
@@ -17,11 +17,7 @@
       }
     end
 
-    it 'fails' do
-      expect do
-        is_expected.to contain_exec('Export /etc/ssl/certs/foo.pfx to /etc/ssl/certs/foo.pem')
-      end.to raise_error(Puppet::Error, %r{either})
-    end
+    it { is_expected.to compile.and_raise_error(%r{either}) }
```

Comment:
@ekohl is that how you have the write in mind? or should we keep the `contain_exec` check? I think that doesn't bring any value if we check for a non-compiling catalog.

Reply:
This is indeed exactly what I was suggesting. You can't really test for resources in a non-compiling catalog.

Reply:
There are a few places with the old syntax, but I wasn't able to convert them properly.